app/uk/gov/hmrc/agentsubscriptionfrontend/models/LinkAccount.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.models
+
+sealed trait LinkAccountAnswer extends Product with Serializable
+
+object LinkAccountAnswer {
+  case object Yes extends LinkAccountAnswer
+  case object No extends LinkAccountAnswer
+
+  def apply(str: String): LinkAccountAnswer = str.toLowerCase match {
```

Comment:
I assume the yes, no are always lower case when submitted using the form. So it should never fail, and if otherwise then should have tests assure correct values are submitted anyway.